### PR TITLE
memory callback test fix

### PR DESCRIPTION
**Pull request by ejohnstown**
At the end of memcb_test() reset the memory allocators before setting them again.

### CODE REVIEW

File: wolfcrypt/test/test.c [hunk 1]
```diff
@@ -19613,6 +19613,7 @@ int memcb_test(void)
 exit_memcb:
 
     /* restore memory callbacks */
+    wolfSSL_ResetAllocators();
```
Comment (toddouska) [inline wolfcrypt/test/test.c]: Is this the right solution?  That is, wouldn't it be easier for users if SetAllocators called ResetAllocators?  Or SetAllocators allowed NULLs?  If this bit us, I think we want to make it easier for users.

**Comment by dgarske** [comment]
I recently added the `wolfSSL_ResetAllocators()` and it has not gone out in a release yet. My preference would be to allow `wolfSSL_SetAllocators()` to pass in NULL's to use the defaults. I decided not to change SetAllocators to avoid breaking any compatibility. Todd if you are okay with it I'd say remove the NULL checks in SetAllocators and remove the new ResetAllocators function.

**Comment by toddouska** [comment]
Allowing SetAllocators to use NULLs seems good, thanks.

**Comment by toddouska** [comment]
And removing ResetAllocators.

**Comment by ejohnstown** [comment]
Sounds good.

**Pull request merged**